extensions/amp-consent/integrating-consent.md

```diff
@@ -0,0 +1,88 @@
+# Integrate your Consent Management Platform (CMP) with AMP Consent (Draft)
+
+## Overview
+
+If you operate a CMP to help publishers to manage their user consents, you may want to integrate your service in `amp-consent`. This will enable your customers to use your service in AMP HTML pages.
+
+## Before you begin
+
+To add your consent management service to AMP runtime, it is expected that you do:
+* Set the remote endpoint to handle requests from AMP. This is for the purpose of determining if the user consent will be required.
+* Make sure the prompted consent collecting page communicates well with the AMP runtime using the provided APIs.
+* Check in your consent configuration to the AMP code base.
+* Be aware of the restrictions that the AMP runtime applies to ensure a good user experience. These includes
+    * Enforce the size of the consent prompt. The only two allowed sizes are the initial size (`100vw`, `30vh`), and the fullScreen after user interactions.
+    * A placeholder will be displayed before the consent prompt iframe is ready.
+    * Enforce the size of the stored consent information. 500kB is the current limit. Please file an issue if you find that not sufficient.
+* Understand that including `<amp-consent type='yourName'></amp-consent>` on the page won't block any components by default. Please make sure to inform your service users to block AMP components either by the `<meta name="amp-consent-blocking>` metaTag, or the `data-block-on-consent` attribute.
+* Create an [Intent-To-Implement issue](../../CONTRIBUTING.md#contributing-features) stating that you'll be adding support to your CMP service to AMP. A great start point is to follow the `_ping_` CMP service implementation that the AMP team creates for testing purpose.
+
+## Add remote endpoint support
+A remote endpoint is expected to tell the AMP runtime whether the user consent is required. It can also pass extra information via AMP to 3P vendors. More information on the remote endpoint can be found [here]((https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/amp-consent.md#checkconsenthref)).
+
+## Prompt UI Iframe
+The AMP runtime will embed the CMP's prompt in an iframe. amp-consent will create the iframe only when it is necessary. A default placeholder will be displayed, and the prompt iframe will remain hidden until it has finished loading.
+
+The prompt iframe and the parent AMP page will communicate through postMessages and the iframe name attribute. In the case of using postMessage, messages from nested iframes will be ignored. The lists of support APIs are:
+
+#### Requesting UI state change
+Iframes can send a `consent-ui` message to the parent AMP page to request UI state change.
+
+``` javascript
+window.parent.postMessage({
+  type: 'consent-ui',
+  action: 'ready/enter-fullscreen'
```

Comment:
I think it may be better, to copy and paste this example, and put each post message type under a bolded word. This will avoid confusion with the slash.

E.g:

**ready**

``` javascript
window.parent.postMessage({
  type: 'consent-ui',
  action: 'ready'
}, '*');
```
Action `'ready'` informs the AMP runtime to hide the placeholder and show the consent prompt instead.